aws/resource_aws_spot_fleet_request.go

```diff
@@ -623,6 +629,13 @@ func resourceAwsSpotFleetRequestCreate(d *schema.ResourceData, meta interface{})
 		spotFleetConfig.AllocationStrategy = aws.String("lowestPrice")
 	}
 
+	if v, ok := d.GetOk("instance_pools_to_use"); ok {
```

Comment:
I would recommend checking against the default value and removing our error check here (letting the API return the appropriate error if necessary), e.g.

```go
if v, ok := d.GetOk("instance_pools_to_use_count"); ok && v.(int) != 1 {
  spotFleetConfig.InstancePoolsToUseCount = aws.Int64(int64(v.(int)))
}
```

Reply:
@bflad could you elaborate on why you prefer to return errors from API rather than catch them earlier? I am no opposing it, just curious about the rationale. 

Working on updating this PR, I see one acceptance test failing due to spot prices jump, not sure if I should alter test spec or ignore it.

Reply:
> could you elaborate on why you prefer to return errors from API rather than catch them earlier? I am not opposing it, just curious about the rationale.

Unless we can generate `plan` time errors (multi-attribute validation in Terraform is a can of worms at the moment), we prefer the API because:
* Upstream API logic may change 
* Upstream API may remove the restriction
* Less code on our end

Which generally boils down to less maintenance burden in this project (someone updating the code then getting a release cut) and for operators (requiring everyone to update their provider). This is the same reasoning why we try to avoid things like EC2 instance type validation, it can be too fast of a moving target or provide little benefit for all the effort.

> Working on updating this PR, I see one acceptance test failing due to spot prices jump, not sure if I should alter test spec or ignore it.

Is it this one? #6031 😄  You can ignore it or potentially look at the forward looking solution mentioned in the comment.

Reply:
Thanks for clarification. Yeah, the PR you mentioned is the exact issue I got. ;-)